src/test/scala/ie/zalando/fabric/gateway/service/IngressDerivationChainSpec.scala

```diff
@@ -664,20 +676,38 @@ class IngressDerivationChainSpec extends FlatSpec with MockitoSugar with Matcher
     allRoutes.filter(isWhitelistedUserRoute) shouldBe empty
   }
 
+  "Cors Support" should "add an options route to each path" in {
```

Comment:
Is this the only test? Maybe we should add in tests to cover all the states for the cors input. enabled v disabled etc...